core/primitives/src/subtrie.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Child trie related struct
+
+use parity_codec::{Encode, Decode};
+use rstd::prelude::*;
+use crate::storage::well_known_keys::CHILD_STORAGE_KEY_PREFIX;
+#[cfg(feature = "std")]
+pub use impl_serde::serialize as bytes;
+
+/// keyspace type.
+pub type KeySpace = Vec<u8>;
```

Comment:
I think the rustdoc here is a bit useless since it is already clearly visible that it is `KeySpace` `type` : )

Can you describe briefly why it is needed, maybe why it is `Vec<u8>`, how it is used and so on.

Reply:
+100. i cannot state strongly enough that:

```
/// foobar type
pub type FooBar = ...;
```

is not only completely unaccenptable in this codebase, but it's almost an insult to the reviewer.